### PR TITLE
Extract popup to component

### PR DESCRIPTION
**Pull request by Chriserus**
# Description

This PR includes extracting Popup as a vue component. It is done in order to make it easier to modify, extend and reuse in the future.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,5 +2,6 @@ node_modules
 dist
 .husky/_
 .husky/.gitignore
+.idea
```
Comment (bokub) [inline .gitignore:5]: Please don't add things related to your IDE...

I suggest you add this to `~/.gitconfig` so you'll never have to do that again:

```
[core]
	excludesFile = ~/.global_gitignore
```

And of course add `.idea` to a new `~/.global_gitignore` file

File: assets/popup.html [hunk 2]
```diff
@@ -4,13 +4,10 @@
...
   <body>
-    <section class="is-hidden" id="alert">
-      <p class="title">The Home Assistant extension in not configured</p>
-      <p class="subtitle"><a href="/options.html" target="_blank">Open options</a></p>
-    </section>
+  <div id="popup"></div>
+  <script src="popup.js"></script>
   </body>
-</html>
+</html>
```
Comment (bokub) [inline assets/popup.html:13]: Did you bypass the git hook?

Please run `npx prettier -w .` to format your code

**Review by bokub — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (Chriserus) [reply .gitignore:5]: Will do! Thank you for the tip!

File: assets/popup.html (hunk 2 above)
Comment (Chriserus) [reply assets/popup.html:13]: Ok, I've done it now, let me know when you finish review, so I'll commit the changes.

**Comment by bokub** [comment]
Style regressions:

## Before / After:
image image

## Before / After:
image
image

**Comment by Chriserus** [comment]
> Style regressions:
> 
> ## Before / After:
> image image
> 
> ## Before / After:
> image
> image

Yes, sorry, I accidently cut out popup.css file, but I already amended the change, you were faster with pulling the code :D

File: .gitignore (hunk 1 above)
Comment (bokub) [reply .gitignore:5]: Go ahead

**Review by bokub — APPROVED** [review]
Looks good to me :+1: 

Thank you!

**Pull request merged**